src/XenCert/StorageHandler.py

```diff
@@ -2478,19 +2478,20 @@ def Create(self):
         sr_ref = None
         try:
             XenCertPrint("First use XAPI to get information for creating an SR.")
-            listSCSIId = []
             (retVal, listAdapters, listSCSIId) = StorageHandlerUtil. \
                                                GetHBAInformation(self.session, \
                                                self.storage_conf, sr_type=self.sr_type)
             if not retVal:                
                 raise Exception("   - Failed to get available HBA information on the host.")
             if len(listSCSIId) == 0:                
                 raise Exception("   - Failed to get available LUNs on the host.")
-
+            avaiableSCSIids = set(listSCSIId) & set(self.storage_conf['scsiIDs'].split(','))
+            if not avaiableSCSIids:
+                raise Exception("   - All LUNs are being used.")
```

Comment:
We would also see this error if the list of scsiIDs the user provided from the CLI was incorrect. I think a better error message would be "None of the specificied SCSI IDs are available. Please confirm that the IDs you provided are valid and that the LUNs are not already in use"

Reply:
Thanks for your suggestion.